src/transports/ssh.c

```diff
@@ -584,6 +584,11 @@ static int _git_ssh_setup_conn(
 		cert_ptr = &cert;
 
 		error = t->owner->certificate_check_cb((git_cert *) cert_ptr, 0, host, t->owner->message_cb_payload);
+
+		/* If the callback chose not to act, fail this hostkey */
+		if (error == GIT_PASSTHROUGH)
+			error = GIT_ECERTIFICATE;
```

Comment:
This is the only location that actually left me wondering. From the documentation, you state that we should behave "as if there was no callback set". But that's not was is happening here, because you do actually abort the connection with a certificate error -- instead, just judging from documentation, we should continue after this `if(certificate_check_cb != NULL)` block

Reply:
Agree; not sure what I was thinking.  Fixed.